transaction-error/src/lib.rs

```diff
@@ -43,8 +46,11 @@ pub enum TransactionError {
     BlockhashNotFound,
 
     /// An error occurred while processing an instruction. The first element of the tuple
-    /// indicates the instruction index in which the error occurred.
-    InstructionError(u8, InstructionError),
+    /// indicates the instruction index in which the error occurred. For avoidance of doubt, the
+    /// third element indicates the address of the program that raised the error, if applicable; the
+    /// error could after all have been raised during a cross-program invocation (ie. in an inner
+    /// instruction).
+    InstructionError(u8, InstructionError, Option<Pubkey>),
```

Comment:
We won't have a program address in cases where the error is `InstructionError::UnsupportedProgramId`, so this is made an `Option`.

Reply:
Yeah, for sure. It's thrown, for instance, here:

https://github.com/anza-xyz/agave/blob/1bcf743f445270407bccf55681f74e06ef8b9a48/program-runtime/src/invoke_context.rs#L250-L252

ie. if you tried to issue an instruction [without providing the address of a program](https://github.com/anza-xyz/agave/blob/1bcf743f445270407bccf55681f74e06ef8b9a48/program-runtime/src/invoke_context.rs#L250-L252).

The larger point here is that there exist `InstructionErrors` that are more like ‘there was a structural problem with this instruction’ rather than ‘a program died and here's why.’ The latter has a program address associated with it while the former may not.